rlclientlib/logger/event_logger.h

```diff
@@ -169,4 +169,28 @@ class slates_logger : public event_logger<slates_decision_event> {
 
     int report_action_taken(const char* event_id, api_status* status);
   };
+
+  class generic_event_logger : public event_logger<generic_event> {
+  public:
+    generic_event_logger(i_message_sender* sender,
+      int send_high_watermark,
```

Comment:
Why the generic logged needs send_high_watermark,send_batch_interval_ms,etc while the old ones don't?

Reply:
My issues is that we're repeating initialization all over the place and this is error prone.

Reply:
They all need - if you look inside cb_logger / ccb_logger /... implementation you will find similar pieces of code which is not visible here. I believe it should be fixed by having non plain config:
"interactions_config":
 { 
    "send_batch_interval": 10, ... 
},
"observations_config":
{
    "send_batch_interval": ...
}
instead of 
{
  "interactions_send_batch_interval": 10, 
  ...
   "observations_send_batch_interval": ...
 ...
}
that we have now.
But it seems independent from what we try to acheve in this PR.

Reply:
I see your point, with configuration sections it becomes easier to pass an uniform configuration container that can be used. 

Right now, that would mean passing a config prefix and depend on conventions. Actually, how bad would this be?

Reply:
maybe it is not that bad, but let's not mix it (it seems independent improvement). I'll create separate issue for that.

Reply:
https://github.com/VowpalWabbit/reinforcement_learning/issues/172